### ISSUE TITLE
NotImplementedError: Don't know how to handle stencil masks when graphics stack depth > 0.

### ISSUE THREAD

**Issue by AEgit** [jbarlow83/OCRmyPDF#90]
When trying to OCR I run into following error:

> NotImplementedError: Don't know how to handle stencil masks when graphics stack depth > 0.

My current workaround is to "print to file", i. e. printing the PDF to another PDF. Thereby, all pages of the original PDF are converted to images. Afterwards the script can be rerun and works fine (the only issue is that the new PDF is, obviously, much larger than the original one). I was just wondering whether:
a) I'm doing something wrong and a similar workaround is NOT needed
b) there's a better workaround
c) if there's no better workaround, whether it would be possible to automate this process (in OCRmyPDF)?

The original PDF can be donwloaded here: https://app.box.com/s/8qv01dvcfjg73utk4ogn8c4gxhnt1rwd
The converted PDF can be downloaded here: https://app.box.com/s/por5n6s8f6b30ye2uhd4pmyv69lryigm
The final PDF (incl. OCR) can be downloaded here: https://app.box.com/s/vylgfb40brcpdj6nff8b6y4q2a2h4q5t

Below you can find the complete error message. 

```
ocrmypdf -f -v -l eng Cox1991.pdf Cox1991_ocr.pdf
  DEBUG - os.symlink(Cox1991.pdf, /tmp/com.github.ocrmypdf.onybvw95/origin)

________________________________________
Tasks which will be run:


Task enters queue = 'ocrmypdf.main.triage' 
  DEBUG - os.symlink(/tmp/com.github.ocrmypdf.onybvw95/origin, /tmp/com.github.ocrmypdf.onybvw95/origin.pdf)
Completed Task = 'ocrmypdf.main.triage' 
Task enters queue = 'ocrmypdf.main.repair_pdf' 
  DEBUG - 
    
    
    
Original exception:

    Exception #1
      'builtins.NotImplementedError(Don't know how to handle stencil masks when graphics stack depth > 0.)' raised in ...
       Task = def ocrmypdf.main.repair_pdf(...):
       Job  = [.../com.github.ocrmypdf.onybvw95/origin.pdf -> .../com.github.ocrmypdf.onybvw95/origin.repaired.pdf, <ocrmypdf.main.WrappedLogger>, [], <unlocked _thread.lock>]
    
    Traceback (most recent call last):
      File "/usr/local/lib/python3.5/dist-packages/ruffus/task.py", line 762, in run_pooled_job_without_exceptions
        params, unglobbed_params)
      File "/usr/lib/python3.5/contextlib.py", line 77, in __exit__
        self.gen.throw(type, value, traceback)
      File "/usr/local/lib/python3.5/dist-packages/ruffus/task.py", line 290, in do_nothing_semaphore
        yield
      File "/usr/local/lib/python3.5/dist-packages/ruffus/task.py", line 751, in run_pooled_job_without_exceptions
        register_cleanup, touch_files_only)
      File "/usr/local/lib/python3.5/dist-packages/ruffus/task.py", line 567, in job_wrapper_io_files
        ret_val = user_defined_work_func(*params)
      File "/usr/local/lib/python3.5/dist-packages/ocrmypdf/main.py", line 509, in repair_pdf
        pdfinfo.extend(pdf_get_all_pageinfo(output_file))
      File "/usr/local/lib/python3.5/dist-packages/ocrmypdf/pageinfo.py", line 354, in pdf_get_all_pageinfo
        return [_pdf_get_pageinfo(infile, n) for n in range(pdf.numPages)]
      File "/usr/local/lib/python3.5/dist-packages/ocrmypdf/pageinfo.py", line 354, in <listcomp>
        return [_pdf_get_pageinfo(infile, n) for n in range(pdf.numPages)]
      File "/usr/local/lib/python3.5/dist-packages/ocrmypdf/pageinfo.py", line 337, in _pdf_get_pageinfo
        page, pageinfo, contentsinfo)]
      File "/usr/local/lib/python3.5/dist-packages/ocrmypdf/pageinfo.py", line 336, in <listcomp>
        pageinfo['images'] = [im for im in _find_page_images(
      File "/usr/local/lib/python3.5/dist-packages/ocrmypdf/pageinfo.py", line 290, in _find_page_images
        yield from _find_page_regular_images(page, pageinfo, contentsinfo)
      File "/usr/local/lib/python3.5/dist-packages/ocrmypdf/pageinfo.py", line 267, in _find_page_regular_images
        "Don't know how to handle "
    NotImplementedError: Don't know how to handle stencil masks when graphics stack depth > 0.
```

**Comment by jbarlow83** [comment]
Fixed in v4.2.4.

It's a bug. I wasn't sure how to handle the situation described by the error message because I didn't have an example PDF to see how other programs handled it and couldn't find a satisfactory description in the reference manual. Based on your PDF I had enough information to figure out how it's supposed to work.

Related to #86.

**Comment by AEgit** [comment]
Thank you very much! I can confirm that this behaviour has been fixed in version 4.2.4.
